### PR TITLE
User identities

### PR DESCRIPTION
**Pull request by dkershaw10**
New user identities property, and associated objectIdentity resource.
New additional examples

This is replacing PR 5631

NOTE:  Changelog is still pending here.

### CODE REVIEW

File: api-reference/beta/api/user-post-users.md [hunk 1]
```diff
@@ -133,6 +139,78 @@ Content-type: application/json
...
+{
+  "displayName": "John Smith",
+  "id": "4c7be08b-361f-41a8-b1ef-1712f7a3dfb2",
+  "passwordProfile" : {
+    "forceChangePasswordNextSignIn": true,
+    "password": null
+  },
+  "identities": [
+    {
+      "signInType":"signInName",
```
Comment (eketo-msft) [inline api-reference/beta/api/user-post-users.md]: Do we want spaces between property and value?
before: "signInType":"signInName"  after: "signInType": "signInName"

File: api-reference/beta/resources/objectidentity.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+doc_type: resourcePageType
+ms.prod: "microsoft-identity-platform"
+author: "davidmu1"
+---
+
+# objectIdentity resource type
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents an identity used to sign in to a user account or other security principals, although this is currently only supported for users. An identity can be provided by Microsoft, by organizations, or by social identity providers such as FaceBook, Google, or Microsoft Account, that are tied to a user account. This enables the user to sign in to the user account with any of those associated identities.
```
Comment (eketo-msft) [inline api-reference/beta/resources/objectidentity.md]: FaceBook should be Facebook.

File: api-reference/beta/api/user-post-users.md [hunk 3]
```diff
@@ -133,6 +139,78 @@ Content-type: application/json
...
+  "truncated": true,
+  "@odata.type": "microsoft.graph.user",
+  "isCollection": true
+} -->
+```http
+HTTP/1.1 201 Created
+Content-type: application/json
+
+{
+  "displayName": "John Smith",
```
Comment (anchandaAADGraph) [inline api-reference/beta/api/user-post-users.md]: We generally don't return response on 201 right?

File: api-reference/beta/api/user-post-users.md [hunk 4]
```diff
@@ -58,12 +58,18 @@ Because the **user** resource supports [extensions](/graph/extensibility-overvie
 >[!NOTE]
 >Federated users created using this API will be forced to sign in every 12 hours by default. For more information about how to change this, see [Exceptions for token lifetimes](https://docs.microsoft.com/azure/active-directory/develop/active-directory-configurable-token-lifetimes#exceptions).
 
+>[!NOTE]
```
Comment (Lauragra) [inline api-reference/beta/api/user-post-users.md]: We have two Note alerts next to each other in this topic, which might dilute the emphasis. Can these be combined into a single note?

File: api-reference/beta/api/user-post-users.md [hunk 5]
```diff
@@ -42,7 +42,7 @@ POST /users
 
 In the request body, supply a JSON representation of [user](../resources/user.md) object.
 
-The following table shows the properties that are required when you create a user.
+The following table shows the properties that are required when you create a user.  However, if creating a user with any **identities**, these requirements are relaxed. For a [B2C local account identity](../resources/objectidentity.md), only a password profile is required; for a social identity, none of these properties is required.
```
Comment (Lauragra) [inline api-reference/beta/api/user-post-users.md]: Will it be clear to the reader what we mean by "identities", or should we provide a link? Suggested revision: "If you're creating a user with any **identities**, not all properties are required. For a [B2C local account identity], only **passwordProfile** is required. For a social identity, no properties are required."

File: api-reference/beta/api/user-post-users.md (hunk 3 above)
Comment (dkershaw10) [reply api-reference/beta/api/user-post-users.md]: Yeah we do.  We must return the service generated id.

File: api-reference/beta/api/user-post-users.md (hunk 4 above)
Comment (dkershaw10) [reply api-reference/beta/api/user-post-users.md]: Yeah, but they are about 2 completely separate things.  I'm open to whatever way you want to organize this though, as long as the info is there.

File: api-reference/beta/api/user-post-users.md (hunk 5 above)
Comment (dkershaw10) [reply api-reference/beta/api/user-post-users.md]: I'd love to provide a link, but **identities** is only a property in a table.  Can't deep link to that.  Happy to make the suggested changes

File: api-reference/beta/api/user-post-users.md [hunk 6]
```diff
@@ -42,7 +42,7 @@ POST /users
 
 In the request body, supply a JSON representation of [user](../resources/user.md) object.
 
-The following table shows the properties that are required when you create a user.
+The following table shows the properties that are required when you create a user. If you're including an **identities** property for the user you're creating, not all properties are required. For a [B2C local account identity](../resources/objectidentity.md), only  **passwordProfile** is required. For a social identity, no properties are required.
```
Comment (Lauragra) [inline api-reference/beta/api/user-post-users.md]: See if the edit I made here helps to clarify?

File: api-reference/beta/api/user-post-users.md (hunk 4 above)
Comment (Lauragra) [reply api-reference/beta/api/user-post-users.md]: I tried making one normal text and the other a Note.

File: api-reference/beta/api/user-update.md [hunk 7]
```diff
@@ -56,6 +58,7 @@ In the request body, supply the values for relevant fields that should be update
 |employeeId|String|The employee identifier assigned to the user by the organization.|
 |givenName|String|The given name (first name) of the user.|
 |hireDate|DateTimeOffset|The hire date of the user. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 would look like this: `'2014-01-01T00:00:00Z'`|
+|identities|[objectIdentity](../resources/objectidentity.md) collection| Represents the identities that can be used to sign in to this user account. An identity can be provided by Microsoft, by organizations, or by social identity providers such as FaceBook, Google, Microsoft Account, and tied to a user account. Any update to **identities** will replace the entire collection.|
```
Comment (Lauragra) [inline api-reference/beta/api/user-update.md]: "Microsoft Account" should be "Microsoft account". But I think you might just mean "Microsoft" as the identity provider? Microsoft account refers to the sign-in credentials that give access to Microsoft services.

File: api-reference/beta/resources/objectidentity.md (hunk 2 above)
Comment (Lauragra) [inline api-reference/beta/resources/objectidentity.md]: Should the "although..." part of the first sentence be called out separately as a Note? Suggested revision:
Represents an identity used to sign in to a user account or other security principal. An identity can be provided by Microsoft, by organizations, or by social identity providers such as Facebook, Google, or Microsoft, that are tied to a user account. This enables the user to sign in to the user account with any of those associated identities.

Note: This resource is currently only supported for users.

File: api-reference/beta/api/user-post-users.md (hunk 6 above)
Comment (dkershaw10) [reply api-reference/beta/api/user-post-users.md]: Yep - LGTM.  Thanks.

File: api-reference/beta/api/user-post-users.md (hunk 4 above)
Comment (dkershaw10) [reply api-reference/beta/api/user-post-users.md]: OK fine with this.

File: api-reference/beta/resources/objectidentity.md (hunk 2 above)
Comment (dkershaw10) [reply api-reference/beta/resources/objectidentity.md]: I'm taking a slightly different approach.  I've removed mention of any other security principals for now.  When we expose this capability, we can update the topic.

**Comment by dkershaw10** [comment]
@Lauragra @CelesteDG This will be ready to merge once the service update goes live.

**Comment by dkershaw10** [comment]
@CelesteDG @Lauragra Service is updated, so this is good to merge.

**Comment by CelesteDG** [comment]
Branch updated to include latest in master. Clean build. Confirmed approval from @Lauragra 
Merging the PR to master.

**Pull request merged**